src/client/app/effects/github.ts

```diff
@@ -109,9 +110,15 @@ export class GitHubEffects {
             }
 
             return this._github.createOrUpdateGist(
-                description, files, gistId, type === GitHub.GitHubActionTypes.SHARE_PUBLIC_GIST
+                    description, files, gistId, type === GitHub.GitHubActionTypes.SHARE_PUBLIC_GIST
             )
-            .map((gist: IGist) => ({ gist: gist, snippetId: id }));
+            .map((gist: IGist) => ({ gist: gist, snippetId: id }))
+            .catch(exception => {
+                if (exception.status === 404) {
```

Comment:
Discussed offline. Keeping `<unknown>` for now, since it doesn't matter for the purposes of the app whether we populate it with this or with the other person's actual login, as the 'Update' won't show either way. Have modified logic to handle more error codes, in case GitHub changes them to be accurate, and have also made it so that if `gistOwnerId` is already set, we don't change it to `<unknown>`.